syntax = "proto3";

package serve;

service ProfileService {
    rpc CallUserProfile(UserProfileRequest) returns (UserProfileResponse){}
    rpc CallMovieProfile(MovieProfileRequest) returns (MovieProfileResponse){}
}

message UserProfileRequest{
    int64 user_id = 1;
}

message UserProfileResponse{
    string f = 1;
    string h = 2;
    double high = 3;
    int32 total = 4;
    double hratings = 5;
}

message MovieProfileRequest{
    int64 movie_id = 1;
}
message MovieProfileResponse{
    double mean =1;
    int32 total_time = 2;
    string title = 3;
    string genres = 4;
}